library/alloc/src/string.rs

```diff
@@ -1628,11 +1628,12 @@ impl String {
         self.vec.clear()
     }
 
-    /// Creates a draining iterator that removes the specified range in the `String`
-    /// and yields the removed `chars`.
+    /// Removes the specified range from the string, returning all removed
+    /// characters as an iterator.
     ///
-    /// Note: The element range is removed even if the iterator is not
-    /// consumed until the end.
+    /// Note: The specified range is removed even if the iterator is not
+    /// consumed until the end, provided that the iterator is dropped (and not
+    /// leaked with `mem::forget`, for example).
```

Comment:
I'd rather delete the paragraph altogether.

Reply:
Sounds good to me, did you want to do that as part of this PR?

Reply:
Yes, I'll delete it then. I'm not sure if it is better to mention the effect of `mem::forget`, but if it is, there are other iterators that should be treated likewise.